### PR TITLE
Alignment part I - Interface and shmem send

### PR DESCRIPTION
**Pull request by rbx**
Adds the alignment interface, and implementation for shmem send case. Other cases (shm receive check, zmq send/recv) ignore the argument for now.

### CODE REVIEW

**Comment by dennisklein** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- fairmq/zeromq/Message.h  4
- fairmq/ofi/Message.cxx  2
- fairmq/shmem/Message.h  4
- test/message/_message.cxx  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/FairRootGroup/FairMQ/pullRequest?prid=5552530&bid=18205534)

File: fairmq/ofi/TransportFactory.cxx [hunk 1]
```diff
@@ -41,11 +41,22 @@ auto TransportFactory::CreateMessage() -> MessagePtr
     return MessagePtr{new Message(&fMemoryResource)};
 }
 
+auto TransportFactory::CreateMessage(Alignment /* alignment */) -> MessagePtr
+{
+    return MessagePtr{new Message(&fMemoryResource)};
```
Comment (dennisklein) [inline fairmq/ofi/TransportFactory.cxx:46]: let's add a comment `//TODO Do not ignore alignment` or so

**Comment by dennisklein** [comment]
See https://cdash.gsi.de/testDetails.php?test=6707455&build=236887

**Comment by rbx** [comment]
> 
> 
> See https://cdash.gsi.de/testDetails.php?test=6707455&build=236887

That assertion checks if the alignment is a power of 2. I test for that in one of our tests. On my system it fires an exception in that case that is caught by the test. But looks like here it is doing the assert.

**Comment by dennisklein** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- fairmq/ofi/Message.cxx  2
- fairmq/zeromq/Message.h  4
- test/message/_message.cxx  3
- fairmq/shmem/Message.h  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/FairRootGroup/FairMQ/pullRequest?prid=5552530&bid=18205534)

**Comment by rbx** [comment]
Updated. The test is now green: https://cdash.gsi.de/testDetails.php?test=6712707&build=236998

demac016 failed test is unrelated and the remaining debian machine is failing to proceed: `Cannot contact lxbk0199.gsi.de: java.lang.InterruptedException`

**Pull request merged**